cmd/config.go

```diff
@@ -531,7 +538,56 @@ func configInitAction(conf *config.Config, gitSpec *git_handler.GitSpec) cli.Act
 		}
 
 		switch conf.ClusterProvider {
-		case util.AWSClusterProvider:
+		case azure_provider.AzureClusterProvider:
+			conf.AwsConfigure = &aws_provider.AwsConfigure{}
+			ac := azure_provider.NewAzureConfigure()
+			asp := azure_provider.NewRawSP()
+
+			if util.IsExists(
+				util.GetHomePath(azure_provider.AzureHomeDir, azure_provider.AzurePrefix+gitSpec.ID+".json"), true) {
+				if err := ac.ReadSPCredentials(gitSpec.ID); err != nil {
+					return err
+				}
+			}
+
+			if c.Bool("azure-service-principle") {
+				if err := json.NewDecoder(os.Stdin).Decode(&asp); err != nil {
+					return fmt.Errorf("unable to deserialize json from stdin: %s", err.Error())
```

Comment:
maybe `json` should be in uppercase?
e.g. `JSON`

Reply:
fixed